subsys/bluetooth/controller/ll_sw/ull_adv_aux.c

```diff
@@ -527,8 +527,50 @@ uint8_t ll_adv_aux_set_count_get(void)
 
 uint8_t ll_adv_aux_set_remove(uint8_t handle)
 {
-	/* TODO: reset/release primary channel and Aux channel PDUs */
-	return 0;
+	struct ll_adv_set *adv;
+	struct lll_adv *lll;
+#if defined(CONFIG_BT_CTLR_ADV_EXT)
+	struct lll_adv_aux *lll_aux;
+#endif/* !CONFIG_BT_CTLR_ADV_EXT */
+#if defined(CONFIG_BT_CTLR_ADV_PERIODIC)
+	struct lll_adv_sync *lll_sync;
+#endif /* CONFIG_BT_CTLR_ADV_PERIODIC */
+
+	/* Get the advertising set instance */
+	adv = ull_adv_is_created_get(handle);
+	if (!adv) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	if (adv->is_enabled) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	lll = &adv->lll;
+		/* Equivalent to lll_adv_data_dequeue(lll) */
+	lll->adv_data.first = lll->adv_data.last = 0;
```

Comment:
This is not necessary. Instead add test for if new adv set can be created and that adv data is sane. Handle any issues during creation.

Reply:
Removed.